### ISSUE TITLE
Location Improvements: EU Formatting

### ISSUE THREAD

**Issue by pattisdr** [ethyca/fides#3611]
### Is your feature request related to a specific problem?

We have a mixture of formats in our PrivacyNoticeRegions. For example, in the us, it is country_state (us_ca) but in the EU, it is countrygroup_country (eu_fr).  As a first round of location improvements, make all eu-prefixed countries just be their country prefix (fr).  Allow filtering on either region or country.

### Describe the solution you'd like

- Switch all eu prefixed countries to just be the country prefix
- Keep API query params the same: GET /privacy_experience?region=fr_idg, or ?region=fr.  
   - Add new filtering requirements: first prioritize matching "fr_idg" - if this doesn't exist, then search for any matching "fr".
   - In other words, allow the API to fall back to a more generic country match if the region match isn't found.
 - We need a data migration for any existing notices, notice histories, and experiences with these region codes.
 - Update out of the box notices to have new country codes

Further context:  https://ethyca.slack.com/archives/C04S7K0T2V7/p1687204241866249

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

**Comment by Roger-Ethyca** [comment]
moving to done
